packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -371,13 +354,20 @@ export class ScopeHoistingPackager {
     // TODO: handle ESM exports of wrapped entry assets...
     let entry = this.bundle.getMainEntry();
     if (entry && !this.wrappedAssets.has(entry.id)) {
+      let hasNamespace = entry.symbols.hasExportSymbol('*');
+
       for (let {
         asset,
         exportAs,
         symbol,
         exportSymbol,
       } of this.bundleGraph.getExportedSymbols(entry)) {
         if (typeof symbol === 'string') {
+          // Only export default if the module has a namespace (e.g. commonjs).
+          if (hasNamespace && exportAs !== '*') {
+            continue;
+          }
```

Comment:
Previously, if a commonjs module was the entry point of a bundle, this could result in invalid identifiers being added to ESM exports, e.g. `exports["foo-bar"] = "foo bar"` (`foo-bar` is not a valid identifier in JS). Now we only export `default` and the importing code uses that as a namespace object.